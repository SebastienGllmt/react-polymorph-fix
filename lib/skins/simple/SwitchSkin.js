'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SwitchSkin = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

// external libraries


// internal utility functions


var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _props = require('../../utils/props');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SwitchSkin = exports.SwitchSkin = function SwitchSkin(props) {
  var theme = props.theme,
      themeId = props.themeId;

  return _react2.default.createElement(
    'div',
    {
      role: 'presentation',
      'aria-hidden': true,
      className: (0, _classnames2.default)([props.className, theme[themeId].root, props.disabled ? theme[themeId].disabled : null, props.checked ? theme[themeId].checked : null]),
      onClick: function onClick(event) {
        if (!props.disabled && props.onChange) {
          props.onChange(!props.checked, event);
        }
      }
    },
    _react2.default.createElement('input', _extends({}, (0, _props.pickDOMProps)(props), {
      className: theme[themeId].input,
      readOnly: true,
      type: 'checkbox'
    })),
    _react2.default.createElement(
      'div',
      {
        className: (0, _classnames2.default)([theme[themeId].switch, props.checked ? theme[themeId].checked : null])
      },
      _react2.default.createElement('span', { className: theme[themeId].thumb })
    ),
    props.label ? _react2.default.createElement(
      'label',
      { className: theme[themeId].label },
      props.label
    ) : null
  );
};
//# sourceMappingURL=SwitchSkin.js.map