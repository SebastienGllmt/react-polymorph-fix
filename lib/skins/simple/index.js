'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SimpleSkins = undefined;

var _SimpleSkins;

var _components = require('../../components');

var _AutocompleteSkin = require('./AutocompleteSkin');

var _BubbleSkin = require('./BubbleSkin');

var _ButtonSkin = require('./ButtonSkin');

var _CheckboxSkin = require('./CheckboxSkin');

var _FormFieldSkin = require('./FormFieldSkin');

var _HeaderSkin = require('./HeaderSkin');

var _InfiniteScrollSkin = require('./InfiniteScrollSkin');

var _InputSkin = require('./InputSkin');

var _LoadingSpinnerSkin = require('./LoadingSpinnerSkin');

var _ModalSkin = require('./ModalSkin');

var _OptionsSkin = require('./OptionsSkin');

var _ProgressBarSkin = require('./ProgressBarSkin');

var _RadioSkin = require('./RadioSkin');

var _SelectSkin = require('./SelectSkin');

var _SwitchSkin = require('./SwitchSkin');

var _TextAreaSkin = require('./TextAreaSkin');

var _TogglerSkin = require('./TogglerSkin');

var _TooltipSkin = require('./TooltipSkin');

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var SimpleSkins = exports.SimpleSkins = (_SimpleSkins = {}, _defineProperty(_SimpleSkins, _components.IDENTIFIERS.AUTOCOMPLETE, _AutocompleteSkin.AutocompleteSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.BUBBLE, _BubbleSkin.BubbleSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.BUTTON, _ButtonSkin.ButtonSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.CHECKBOX, _CheckboxSkin.CheckboxSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.FORM_FIELD, _FormFieldSkin.FormFieldSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.HEADER, _HeaderSkin.HeaderSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.INFINITE_SCROLL, _InfiniteScrollSkin.InfiniteScrollSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.INPUT, _InputSkin.InputSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.LOADING_SPINNER, _LoadingSpinnerSkin.LoadingSpinnerSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.MODAL, _ModalSkin.ModalSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.OPTIONS, _OptionsSkin.OptionsSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.PROGRESS_BAR, _ProgressBarSkin.ProgressBarSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.RADIO, _RadioSkin.RadioSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.SELECT, _SelectSkin.SelectSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.SWITCH, _SwitchSkin.SwitchSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TEXT_AREA, _TextAreaSkin.TextAreaSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TOGGLER, _TogglerSkin.TogglerSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TOOLTIP, _TooltipSkin.TooltipSkin), _SimpleSkins);
//# sourceMappingURL=index.js.map