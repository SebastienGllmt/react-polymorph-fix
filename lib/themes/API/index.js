'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.IDENTIFIERS = exports.ROOT_THEME_API = undefined;

var _ROOT_THEME_API;

var _components = require('../../components');

var _autocomplete = require('./autocomplete');

var _bubble = require('./bubble');

var _button = require('./button');

var _checkbox = require('./checkbox');

var _flex = require('./flex');

var _formfield = require('./formfield');

var _grid = require('./grid');

var _gutter = require('./gutter');

var _header = require('./header');

var _infinitescroll = require('./infinitescroll');

var _input = require('./input');

var _loadingspinner = require('./loadingspinner');

var _modal = require('./modal');

var _options = require('./options');

var _progressbar = require('./progressbar');

var _radio = require('./radio');

var _select = require('./select');

var _switch = require('./switch');

var _textarea = require('./textarea');

var _toggler = require('./toggler');

var _tooltip = require('./tooltip');

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var ROOT_THEME_API = exports.ROOT_THEME_API = (_ROOT_THEME_API = {}, _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.AUTOCOMPLETE, _autocomplete.AUTOCOMPLETE_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.BUBBLE, _bubble.BUBBLE_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.BUTTON, _button.BUTTON_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.CHECKBOX, _checkbox.CHECKBOX_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.FLEX, _flex.FLEX_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.FORM_FIELD, _formfield.FORM_FIELD_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.GRID, _grid.GRID_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.GUTTER, _gutter.GUTTER_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.HEADER, _header.HEADER_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.INFINITE_SCROLL, _infinitescroll.INFINITE_SCROLL_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.INPUT, _input.INPUT_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.LOADING_SPINNER, _loadingspinner.LOADING_SPINNER_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.MODAL, _modal.MODAL_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.OPTIONS, _options.OPTIONS_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.PROGRESS_BAR, _progressbar.PROGRESS_BAR_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.RADIO, _radio.RADIO_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.SELECT, _select.SELECT_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.SWITCH, _switch.SWITCH_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.TEXT_AREA, _textarea.TEXT_AREA_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.TOGGLER, _toggler.TOGGLER_THEME_API), _defineProperty(_ROOT_THEME_API, _components.IDENTIFIERS.TOOLTIP, _tooltip.TOOLTIP_THEME_API), _ROOT_THEME_API);

exports.IDENTIFIERS = _components.IDENTIFIERS;
//# sourceMappingURL=index.js.map