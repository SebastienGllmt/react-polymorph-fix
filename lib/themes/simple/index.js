'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SimpleTheme = undefined;

var _SimpleTheme;

var _components = require('../../components');

var _SimpleAutocomplete = require('./SimpleAutocomplete.scss');

var _SimpleAutocomplete2 = _interopRequireDefault(_SimpleAutocomplete);

var _SimpleBubble = require('./SimpleBubble.scss');

var _SimpleBubble2 = _interopRequireDefault(_SimpleBubble);

var _SimpleButton = require('./SimpleButton.scss');

var _SimpleButton2 = _interopRequireDefault(_SimpleButton);

var _SimpleCheckbox = require('./SimpleCheckbox.scss');

var _SimpleCheckbox2 = _interopRequireDefault(_SimpleCheckbox);

var _SimpleFlex = require('./SimpleFlex.scss');

var _SimpleFlex2 = _interopRequireDefault(_SimpleFlex);

var _SimpleFormField = require('./SimpleFormField.scss');

var _SimpleFormField2 = _interopRequireDefault(_SimpleFormField);

var _SimpleGrid = require('./SimpleGrid.scss');

var _SimpleGrid2 = _interopRequireDefault(_SimpleGrid);

var _SimpleGutter = require('./SimpleGutter.scss');

var _SimpleGutter2 = _interopRequireDefault(_SimpleGutter);

var _SimpleHeader = require('./SimpleHeader.scss');

var _SimpleHeader2 = _interopRequireDefault(_SimpleHeader);

var _SimpleInfiniteScroll = require('./SimpleInfiniteScroll.scss');

var _SimpleInfiniteScroll2 = _interopRequireDefault(_SimpleInfiniteScroll);

var _SimpleInput = require('./SimpleInput.scss');

var _SimpleInput2 = _interopRequireDefault(_SimpleInput);

var _SimpleLoadingSpinner = require('./SimpleLoadingSpinner.scss');

var _SimpleLoadingSpinner2 = _interopRequireDefault(_SimpleLoadingSpinner);

var _SimpleModal = require('./SimpleModal.scss');

var _SimpleModal2 = _interopRequireDefault(_SimpleModal);

var _SimpleOptions = require('./SimpleOptions.scss');

var _SimpleOptions2 = _interopRequireDefault(_SimpleOptions);

var _SimpleProgressBar = require('./SimpleProgressBar.scss');

var _SimpleProgressBar2 = _interopRequireDefault(_SimpleProgressBar);

var _SimpleRadio = require('./SimpleRadio.scss');

var _SimpleRadio2 = _interopRequireDefault(_SimpleRadio);

var _SimpleSelect = require('./SimpleSelect.scss');

var _SimpleSelect2 = _interopRequireDefault(_SimpleSelect);

var _SimpleSwitch = require('./SimpleSwitch.scss');

var _SimpleSwitch2 = _interopRequireDefault(_SimpleSwitch);

var _SimpleTextArea = require('./SimpleTextArea.scss');

var _SimpleTextArea2 = _interopRequireDefault(_SimpleTextArea);

var _SimpleToggler = require('./SimpleToggler.scss');

var _SimpleToggler2 = _interopRequireDefault(_SimpleToggler);

var _SimpleTooltip = require('./SimpleTooltip.scss');

var _SimpleTooltip2 = _interopRequireDefault(_SimpleTooltip);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
// import theme IDENTIFIERS constants


// css modules plugin converts all imports below into plain objects


// SimpleTheme is a plain object serving as the default export.
// Each key is named after a component and each key's value
// is the component's corresponding theme. The user can
// pass this entire obj directly to ThemeProvider via the "theme" prop
var SimpleTheme = exports.SimpleTheme = (_SimpleTheme = {}, _defineProperty(_SimpleTheme, _components.IDENTIFIERS.AUTOCOMPLETE, _SimpleAutocomplete2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.BUBBLE, _SimpleBubble2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.BUTTON, _SimpleButton2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.CHECKBOX, _SimpleCheckbox2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.FLEX, _SimpleFlex2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.FORM_FIELD, _SimpleFormField2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.GRID, _SimpleGrid2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.GUTTER, _SimpleGutter2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.HEADER, _SimpleHeader2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.INFINITE_SCROLL, _SimpleInfiniteScroll2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.INPUT, _SimpleInput2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.LOADING_SPINNER, _SimpleLoadingSpinner2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.MODAL, _SimpleModal2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.OPTIONS, _SimpleOptions2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.PROGRESS_BAR, _SimpleProgressBar2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.RADIO, _SimpleRadio2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.SELECT, _SimpleSelect2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.SWITCH, _SimpleSwitch2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.TEXT_AREA, _SimpleTextArea2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.TOGGLER, _SimpleToggler2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.TOOLTIP, _SimpleTooltip2.default), _SimpleTheme);
//# sourceMappingURL=index.js.map