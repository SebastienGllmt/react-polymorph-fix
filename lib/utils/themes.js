'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.didThemePropsChange = exports.composeTheme = exports.addThemeId = exports.composeComponentStyles = exports.appendToProperty = undefined;

var _isEqual2 = require('lodash/isEqual');

var _isEqual3 = _interopRequireDefault(_isEqual2);

var _isEmpty2 = require('lodash/isEmpty');

var _isEmpty3 = _interopRequireDefault(_isEmpty2);

var _cloneDeep2 = require('lodash/cloneDeep');

var _cloneDeep3 = _interopRequireDefault(_cloneDeep2);

var _typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol" ? function (obj) { return typeof obj; } : function (obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; };

var _props = require('./props');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var appendToProperty = exports.appendToProperty = function appendToProperty(dest, name, value) {
  dest[name] === '' ? dest[name] = value : dest[name] += ' ' + value;
};

var composeComponentStyles = exports.composeComponentStyles = function composeComponentStyles(componentStyles, componentTheme) {
  if (!componentTheme) return;
  for (var property in componentStyles) {
    if ((0, _props.hasProperty)(componentStyles, property)) {
      if ((0, _props.hasProperty)(componentTheme, property)) {
        appendToProperty(componentStyles, property, componentTheme[property]);
      }
    }
  }
};

// checks for the existence of a property on theme
// that matches the value of themeId (string)
// if the property exists, also checks the type of
// theme[themeId] to ensure it's an object
var addThemeId = exports.addThemeId = function addThemeId() {
  var theme = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var themeId = arguments[1];

  if (theme && !(0, _isEmpty3.default)(theme) && themeId) {
    var themeIdExists = (0, _props.hasProperty)(theme, themeId);
    var themeIdIsObj = _typeof(theme[themeId]) === 'object';
    return themeIdExists && themeIdIsObj ? theme : _defineProperty({}, themeId, theme);
  }
  return theme;
};

/**
 * Composes a base theme with the given overrides, which should
 * be provided in the same schema, defined by the theme API param.
 *
 * @param theme - The base theme to be composed with overrides
 * @param themeOverrides - The custom overrides for the base theme
 * @param themeAPI - The theme API schema that should be used for composition
 * @returns {{}} - The composed theme
 */

var composeTheme = exports.composeTheme = function composeTheme() {
  var theme = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var themeOverrides = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};
  var themeAPI = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : {};

  // Return theme if there are no overrides provided
  if ((0, _isEmpty3.default)(themeOverrides)) return theme;

  // final object to be returned
  var composedTheme = (0, _cloneDeep3.default)(themeAPI);

  for (var componentId in themeAPI) {
    if ((0, _props.hasProperty)(composedTheme, componentId)) {
      var componentStyles = composedTheme[componentId];
      composeComponentStyles(componentStyles, theme[componentId]);
      composeComponentStyles(componentStyles, themeOverrides[componentId]);
    }
  }
  return composedTheme;
};

// Used in componentWillReceiveProps, this function compares the current
// set of theme related props against the next set to see if any have changed.
// If true, a component's theme is recomposed and local state is updated
var didThemePropsChange = exports.didThemePropsChange = function didThemePropsChange(_ref2, _ref3, setState) {
  var context = _ref2.context,
      themeId = _ref2.themeId,
      theme = _ref2.theme,
      themeOverrides = _ref2.themeOverrides;
  var nextContext = _ref3.context,
      nextThemeId = _ref3.themeId,
      nextTheme = _ref3.theme,
      nextOverrides = _ref3.themeOverrides;

  if (!(0, _isEqual3.default)(context, nextContext) || !(0, _isEqual3.default)(themeId, nextThemeId) || !(0, _isEqual3.default)(theme, nextTheme) || !(0, _isEqual3.default)(themeOverrides, nextOverrides)) {
    setState(function () {
      return {
        composedTheme: composeTheme(addThemeId(nextTheme || nextContext.theme, nextThemeId), addThemeId(nextOverrides, nextThemeId), nextContext.ROOT_THEME_API)
      };
    });
  }
};
//# sourceMappingURL=themes.js.map