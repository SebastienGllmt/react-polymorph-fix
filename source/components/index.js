// @flow
export const IDENTIFIERS = {
  AUTOCOMPLETE: 'autocomplete',
  BUBBLE: 'bubble',
  BUTTON: 'button',
  CHECKBOX: 'checkbox',
  FLEX: 'flex',
  FORM_FIELD: 'formfield',
  GRID: 'grid',
  GUTTER: 'gutter',
  HEADER: 'header',
  INFINITE_SCROLL: 'infinitescroll',
  INPUT: 'input',
  LOADING_SPINNER: 'loadingspinner',
  MODAL: 'modal',
  OPTIONS: 'options',
  PROGRESS_BAR: 'progressbar',
  RADIO: 'radio',
  SELECT: 'select',
  SWITCH: 'switch',
  TEXT_AREA: 'textarea',
  TOGGLER: 'toggler',
  TOOLTIP: 'tooltip'
};
