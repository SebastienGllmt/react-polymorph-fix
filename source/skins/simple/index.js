// @flow
import { IDENTIFIERS } from '../../components';
import { AutocompleteSkin } from './AutocompleteSkin';
import { BubbleSkin } from './BubbleSkin';
import { ButtonSkin } from './ButtonSkin';
import { CheckboxSkin } from './CheckboxSkin';
import { FormFieldSkin } from './FormFieldSkin';
import { HeaderSkin } from './HeaderSkin';
import { InfiniteScrollSkin } from './InfiniteScrollSkin';
import { InputSkin } from './InputSkin';
import { LoadingSpinnerSkin } from './LoadingSpinnerSkin';
import { ModalSkin } from './ModalSkin';
import { OptionsSkin } from './OptionsSkin';
import { ProgressBarSkin } from './ProgressBarSkin';
import { RadioSkin } from './RadioSkin';
import { SelectSkin } from './SelectSkin';
import { SwitchSkin } from './SwitchSkin';
import { TextAreaSkin } from './TextAreaSkin';
import { TogglerSkin } from './TogglerSkin';
import { TooltipSkin } from './TooltipSkin';

export const SimpleSkins = {
  [IDENTIFIERS.AUTOCOMPLETE]: AutocompleteSkin,
  [IDENTIFIERS.BUBBLE]: BubbleSkin,
  [IDENTIFIERS.BUTTON]: ButtonSkin,
  [IDENTIFIERS.CHECKBOX]: CheckboxSkin,
  [IDENTIFIERS.FORM_FIELD]: FormFieldSkin,
  [IDENTIFIERS.HEADER]: HeaderSkin,
  [IDENTIFIERS.INFINITE_SCROLL]: InfiniteScrollSkin,
  [IDENTIFIERS.INPUT]: InputSkin,
  [IDENTIFIERS.LOADING_SPINNER]: LoadingSpinnerSkin,
  [IDENTIFIERS.MODAL]: ModalSkin,
  [IDENTIFIERS.OPTIONS]: OptionsSkin,
  [IDENTIFIERS.PROGRESS_BAR]: ProgressBarSkin,
  [IDENTIFIERS.RADIO]: RadioSkin,
  [IDENTIFIERS.SELECT]: SelectSkin,
  [IDENTIFIERS.SWITCH]: SwitchSkin,
  [IDENTIFIERS.TEXT_AREA]: TextAreaSkin,
  [IDENTIFIERS.TOGGLER]: TogglerSkin,
  [IDENTIFIERS.TOOLTIP]: TooltipSkin
};
