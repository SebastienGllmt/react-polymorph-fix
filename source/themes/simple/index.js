// @flow
// import theme IDENTIFIERS constants
import { IDENTIFIERS } from '../../components';

// css modules plugin converts all imports below into plain objects
import SimpleAutocomplete from './SimpleAutocomplete.scss';
import SimpleBubble from './SimpleBubble.scss';
import SimpleButton from './SimpleButton.scss';
import SimpleCheckbox from './SimpleCheckbox.scss';
import SimpleFlex from './SimpleFlex.scss';
import SimpleFormField from './SimpleFormField.scss';
import SimpleGrid from './SimpleGrid.scss';
import SimpleGutter from './SimpleGutter.scss';
import SimpleHeader from './SimpleHeader.scss';
import SimpleInfiniteScroll from './SimpleInfiniteScroll.scss';
import SimpleInput from './SimpleInput.scss';
import SimpleLoadingSpinner from './SimpleLoadingSpinner.scss';
import SimpleModal from './SimpleModal.scss';
import SimpleOptions from './SimpleOptions.scss';
import SimpleProgressBar from './SimpleProgressBar.scss';
import SimpleRadio from './SimpleRadio.scss';
import SimpleSelect from './SimpleSelect.scss';
import SimpleSwitch from './SimpleSwitch.scss';
import SimpleTextArea from './SimpleTextArea.scss';
import SimpleToggler from './SimpleToggler.scss';
import SimpleTooltip from './SimpleTooltip.scss';

// SimpleTheme is a plain object serving as the default export.
// Each key is named after a component and each key's value
// is the component's corresponding theme. The user can
// pass this entire obj directly to ThemeProvider via the "theme" prop
export const SimpleTheme = {
  [IDENTIFIERS.AUTOCOMPLETE]: SimpleAutocomplete,
  [IDENTIFIERS.BUBBLE]: SimpleBubble,
  [IDENTIFIERS.BUTTON]: SimpleButton,
  [IDENTIFIERS.CHECKBOX]: SimpleCheckbox,
  [IDENTIFIERS.FLEX]: SimpleFlex,
  [IDENTIFIERS.FORM_FIELD]: SimpleFormField,
  [IDENTIFIERS.GRID]: SimpleGrid,
  [IDENTIFIERS.GUTTER]: SimpleGutter,
  [IDENTIFIERS.HEADER]: SimpleHeader,
  [IDENTIFIERS.INFINITE_SCROLL]: SimpleInfiniteScroll,
  [IDENTIFIERS.INPUT]: SimpleInput,
  [IDENTIFIERS.LOADING_SPINNER]: SimpleLoadingSpinner,
  [IDENTIFIERS.MODAL]: SimpleModal,
  [IDENTIFIERS.OPTIONS]: SimpleOptions,
  [IDENTIFIERS.PROGRESS_BAR]: SimpleProgressBar,
  [IDENTIFIERS.RADIO]: SimpleRadio,
  [IDENTIFIERS.SELECT]: SimpleSelect,
  [IDENTIFIERS.SWITCH]: SimpleSwitch,
  [IDENTIFIERS.TEXT_AREA]: SimpleTextArea,
  [IDENTIFIERS.TOGGLER]: SimpleToggler,
  [IDENTIFIERS.TOOLTIP]: SimpleTooltip
};
