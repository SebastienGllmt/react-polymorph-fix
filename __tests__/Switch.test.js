import React from 'react';
import renderer from 'react-test-renderer';

import { Checkbox } from '../source/components/Checkbox';
import { SwitchSkin } from '../source/skins/simple/SwitchSkin';
import { IDENTIFIERS } from '../source/themes/API';
import { renderInSimpleTheme } from './helpers/theming';

test('Switch renders correctly', () => {
  const component = renderInSimpleTheme(
    <Checkbox
      themeId={IDENTIFIERS.SWITCH}
      skin={SwitchSkin}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Switch renders with label', () => {
  const component = renderInSimpleTheme(
    <Checkbox
      themeId={IDENTIFIERS.SWITCH}
      label="click here"
      skin={SwitchSkin}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Switch is disabled', () => {
  const component = renderInSimpleTheme(
    <Checkbox
      themeId={IDENTIFIERS.SWITCH}
      disabled
      skin={SwitchSkin}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Switch is disabled and renders a label', () => {
  const component = renderInSimpleTheme(
    <Checkbox
      themeId={IDENTIFIERS.SWITCH}
      disabled
      label="click here"
      skin={SwitchSkin}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Switch is checked', () => {
  const component = renderInSimpleTheme(
    <Checkbox
      themeId={IDENTIFIERS.SWITCH}
      checked
      skin={SwitchSkin}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
