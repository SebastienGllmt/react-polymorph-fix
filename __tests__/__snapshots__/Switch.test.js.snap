// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch is checked 1`] = `
<div
  aria-hidden={true}
  className="root checked"
  onClick={[Function]}
  role="presentation"
>
  <input
    checked={true}
    className="input"
    readOnly={true}
    type="checkbox"
  />
  <div
    className="switch checked"
  >
    <span
      className="thumb"
    />
  </div>
</div>
`;

exports[`Switch is disabled 1`] = `
<div
  aria-hidden={true}
  className="root disabled"
  onClick={[Function]}
  role="presentation"
>
  <input
    checked={false}
    className="input"
    disabled={true}
    readOnly={true}
    type="checkbox"
  />
  <div
    className="switch"
  >
    <span
      className="thumb"
    />
  </div>
</div>
`;

exports[`Switch is disabled and renders a label 1`] = `
<div
  aria-hidden={true}
  className="root disabled"
  onClick={[Function]}
  role="presentation"
>
  <input
    checked={false}
    className="input"
    disabled={true}
    label="click here"
    readOnly={true}
    type="checkbox"
  />
  <div
    className="switch"
  >
    <span
      className="thumb"
    />
  </div>
  <label
    className="label"
  >
    click here
  </label>
</div>
`;

exports[`Switch renders correctly 1`] = `
<div
  aria-hidden={true}
  className="root"
  onClick={[Function]}
  role="presentation"
>
  <input
    checked={false}
    className="input"
    readOnly={true}
    type="checkbox"
  />
  <div
    className="switch"
  >
    <span
      className="thumb"
    />
  </div>
</div>
`;

exports[`Switch renders with label 1`] = `
<div
  aria-hidden={true}
  className="root"
  onClick={[Function]}
  role="presentation"
>
  <input
    checked={false}
    className="input"
    label="click here"
    readOnly={true}
    type="checkbox"
  />
  <div
    className="switch"
  >
    <span
      className="thumb"
    />
  </div>
  <label
    className="label"
  >
    click here
  </label>
</div>
`;
